### PR TITLE
Command to update Addressbase and UprnToCouncil data

### PR DESCRIPTION
**Pull request by GeoWill**
Depends on: https://github.com/DemocracyClub/uk-geo-utils/pull/30

To test it do a local checkout of the this branch ~~and the uk-geo-utils PR, and then `pip install -e path/to/uk-geo-utils`~~ .
You can then run:
```shell
./manage.py update_addressbase --addressbase-path /home/will/Downloads/addressbase-2024-08-06/addressbase
_cleaned.csv --uprntocouncil-path /home/will/Downloads/addressbase-2024-08-06/uprn-to-councils.csv
```
The files above are available on s3 in the private data bucket on the prod account. 

To make iteration cycles faster it's worth making the files shorter, however to make it work you need to have matching uprns.[1]

I've also added these to the same bucket under the `addressbase/sample/` key. 
These can be used for testing the downloads:
```shell
    ./manage.py update_addressbase \
        --addressbase-s3-uri='s3://bucket/addressbase/sample/addressbase_cleaned/addressbase_cleaned.csv' \
        --uprntocouncil-s3-uri='s3://bucket/addressbase/sample/uprn-to-council/uprn-to-councils.csv'
```

~~At this stage this still needs to be tested in the dev environment against RDS.~~

This is now done it took a t3.2xlarge rds just under 17m. The time spent on the transaction was short. Not sure exactly how long though.

Assuming that works the process for updating addressbase would look like:
 * Download new addressbase from OS
 * create cleaned_addressbase.csv
 * Import that locally, and create the uprn-to-council.csv file.
 * Upload both these csvs to s3
 * ssh into an EC2 instance and download the csvs
 * run this command. 

[1] I used [miller](https://miller.readthedocs.io) to do this:

```shell
mlr --csv --hi --ho  filter '$1 % 599 == 0' uprn-to-councils.csv > uprn-to-councils-reduced.csv
mlr --csv --hi --ho  filter '$1 % 599 == 0' addressbase_cleaned.csv > addressbase_cleaned-reduced.csv
```

### CODE REVIEW

**Comment by GeoWill** [comment]
Run teardown as the last thing inside the transaction

File: polling_stations/apps/addressbase/management/commands/update_addressbase.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+                # Rename temp tables
+                addressbase_updater.rename_temp_table()
+                uprntocouncil_updater.rename_temp_table()
+
+            # Add foreign keys back
+            if addressbase_updater.foreign_key_constraints:
+                addressbase_updater.add_foreign_keys()
+            if uprntocouncil_updater.foreign_key_constraints:
+                uprntocouncil_updater.add_foreign_keys()
```
[the remark this answers is not in the document]
Comment (GeoWill) [reply polling_stations/apps/addressbase/management/commands/update_addressbase.py]: Figured it out: https://github.com/DemocracyClub/uk-geo-utils/pull/30/commits/dd3f763697c833ba23dadc6092e124de97c48508#diff-6c87a9f5ed277ffa23f05fbaa0d0d4d164d6379ddaf481f414e470a24de16915R208

File: polling_stations/apps/addressbase/management/commands/update_addressbase.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                self.style.SUCCESS(
+                    "Successfully updated both Addressbase and UPRN to Council tables"
+                )
+            )
+
+        finally:
+            # Clean up both updaters
+            self.stdout.write("Cleaning up...")
+            addressbase_updater.db_cleanup()
+            uprntocouncil_updater.db_cleanup()
```
Comment (chris48s) [inline polling_stations/apps/addressbase/management/commands/update_addressbase.py:167]: If we downloaded the files from S3 and saved them to temp, can we clean up the temp files in the finally block too?

**Comment by chris48s** [comment]
Can we make updating the docs in https://github.com/DemocracyClub/UK-Polling-Stations/wiki/Keeping-addressbase-up-to-date a follow up to this PR?

**Comment by chris48s** [comment]
Any idea what happened with the latest build failure?

**Comment by chris48s** [comment]
OK, so I've done a bit of digging on why the build is failing.. I can't really explain why, but the commit hash that `get_last_import_sha_from_ssm()` is picking up is a commit from a branch that has been force pushed over.

So we're trying to run

```
git diff --name-only e7dc69d5628a19ca314797e24e74853b14ff96fe 3b11ee3fb3dee9d5064d38824a1e5ca5597d57dc
```

but that's failing because e7dc69d5628a19ca314797e24e74853b14ff96fe isn't a commit that exists in the tree. The equivalent commit that got merged to master was 2e99090f900b0f1ff312940430ac196952a53586

I don't really understand _why_ that's happening though. Seems odd given how far back in the history that is.

**Comment by chris48s** [comment]
The latest change look sensible.

I'm really confused about why those latest commits have any bearing on what `get_last_import_sha_from_ssm()` returns :confused:

**Review by chris48s — APPROVED** [review]

**Comment by chris48s** [comment]
Just for code archaeology purposes, the failing "Development: Run New Imports Post Deploy" job was running on deploy to dev not this branch (we'd pushed the same commit to both this branch and the development branch).
If we've got ourselves into that state by force-pushing the development branch, we can manually change the `LAST_IMPORT_SHA` variable in the relevant AWS account to un-stick it.

**Pull request merged**